src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -697,14 +706,60 @@ impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
             None
         };
 
-        let volume_mounts = if self.config.coverage {
-            Some(vec![VolumeMount {
+        let mut volume_mounts = vec![];
+
+        if self.config.coverage {
+            volume_mounts.push(VolumeMount {
                 name: "coverage".to_string(),
                 mount_path: "/coverage".to_string(),
                 ..Default::default()
-            }])
-        } else {
-            None
+            })
+        }
+
+        let volumes = match (disk, &self.config.ephemeral_volume_storage_class) {
+            (true, Some(ephemeral_volume_storage_class)) => {
+                volume_mounts.push(VolumeMount {
+                    name: "scratch-volume".to_string(),
+                    mount_path: "/scratch-directory".to_string(),
+                    ..Default::default()
+                });
+                args.push("--scratch-directory=/scratch-directory".into());
+
+                Some(vec![Volume {
+                    name: "scratch-volume".to_string(),
```

Comment:
I think it's cleaner not to include the type in the name. Doesn't need to be a volume named "scratch-volume"; can just be a volume named "scratch".

```suggestion
                volume_mounts.push(VolumeMount {
                    name: "scratch".to_string(),
                    mount_path: "/scratch".to_string(),
                    ..Default::default()
                });
                args.push("--scratch-directory=/scratch".into());

                Some(vec![Volume {
                    name: "scratch".to_string(),
```